src/main/scala/beam/router/R5RoutingApp.scala

```diff
@@ -0,0 +1,184 @@
+    for {
+      x <- c.downField("x").as[String]
+      y <- c.downField("y").as[String]
+    } yield {
+      new Location(x.toDouble, y.toDouble)
+    }
+  }
+}
+
+class Format[T](implicit decode: Lazy[DerivedDecoder[T]], encode: Lazy[DerivedObjectEncoder[T]])
+    extends Encoder[T]
+    with Decoder[T] {
+  implicit val decoder: Decoder[T] = deriveDecoder[T]
+  implicit val encoder: Encoder[T] = deriveEncoder[T]
+
+  override def apply(a: T): Json = encoder.apply(a)
+
+  override def apply(c: HCursor): Result[T] = decoder.apply(c)
+}
+
+object AllNeededFormats {
+  implicit val locationFormat = LocationFormat
+  implicit val beamModeFormat = new Format[BeamMode]
+  implicit val vehicleIdFormat = new IdFormat[Vehicle]
+  implicit val beamVehicleTypeFormat = new IdFormat[BeamVehicleType]
+
+  implicit val spaceTimeFormat = new Format[SpaceTime]
+  implicit val streetVehicleFormat = new Format[StreetVehicle]
+  implicit val householdAttributesFormat = new Format[HouseholdAttributes]
+  implicit val attributesOfIndividualFormat = new Format[AttributesOfIndividual]
+  implicit val intermodalUseFormat = new Format[IntermodalUse]
+  implicit val routingRequestFormat = new Format[RoutingRequest]
+
+  implicit val transitStopsInfoFormat = new Format[TransitStopsInfo]
+  implicit val beamPathFormat = new Format[BeamPath]
+  implicit val beamLegFormat = new Format[BeamLeg]
+  implicit val embodiedBeamLegFormat = new Format[EmbodiedBeamLeg]
+  implicit val embodiedBeamTripFormat = new Format[EmbodiedBeamTrip]
+  implicit val routingResponseFormat = new Format[RoutingResponse]
+}
+
+class RoutingHandler(val workerRouter: ActorRef) extends FailFastCirceSupport {
+  implicit val timeout: Timeout = new Timeout(10, TimeUnit.SECONDS)
+  import AllNeededFormats._
+
+  val route: Route = {
+    path("find-route") {
+      post {
+        entity(as[RoutingRequest]) { request =>
+          complete(workerRouter.ask(request).mapTo[RoutingResponse])
+        }
+      }
+    }
+  }
+}
+
+object CustomExceptionHandling extends LazyLogging {
+
+  def handler: ExceptionHandler = ExceptionHandler {
+    case t: Throwable =>
+      extractClientIP { remoteAddress =>
+        extractRequest { request =>
+          val msg = s"Exception during processing $request from $remoteAddress: ${t.getMessage}"
+          logger.error(msg, t)
+          complete(HttpResponse(StatusCodes.InternalServerError, entity = msg))
+        }
+      }
+  }
+}
+
+object R5RoutingApp extends BeamHelper {
+  import AllNeededFormats._
+
+  implicit val actorSystem: ActorSystem = ActorSystem()
+  implicit val materializer: ActorMaterializer = ActorMaterializer()
+  implicit val timeout: Timeout = new Timeout(600, TimeUnit.SECONDS)
+
+  def main(args: Array[String]): Unit = {
+    val (arg, cfg) = prepareConfig(args, isConfigArgRequired = true)
+    val workerRouter: ActorRef = actorSystem.actorOf(Props(classOf[R5RoutingWorker], cfg), name = "workerRouter")
```

Comment:
What is the purpose of this app? Would it be useful to keep this primed?

Reply:
I use it to investigate the travel time for some routing requests. Currently there is no way to do it in Beam (as far as I know).